optuna/storages/_rdb/storage.py

```diff
@@ -1132,6 +1133,50 @@ def get_all_versions(self) -> List[str]:
 
         return self._version_manager.get_all_versions()
 
+    async def record_heartbeat(self, trial_id: int, interval: int) -> None:
+        """Record the heartbeat (timestamp) of the trial.
+
+        This method is specific to RDBStorage.
+
+        Args:
+            trial_id:
+                ID of the trial.
+            interval:
+                Interval to record the heartbeat. It is recorded every ``interval`` seconds.
+        """
+
+        while True:
+            with _create_scoped_session(self.scoped_session, True) as session:
+                step = models.TrialTimeStampModel.next_step(trial_id, session)
+                timestamp = models.TrialTimeStampModel(trial_id=trial_id, step=step)
+                session.add(timestamp)
+            await asyncio.sleep(interval)
```

Comment:
Can we do this at the caller instead of inside the storage? Propagating event loops to the storage increases code complexity quite a bit.